stock_reorder_forecast/wizards/purchase_wizard.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import date, datetime
+from openerp import api, fields, models
+from openerp.tools import float_round
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DSDF
+
+
+class PurchaseWizard(models.TransientModel):
+
+    _name = "purchase.purchase_wizard"
+    _description = "wizard create proposal purchase"
+
+    @api.model
+    def default_get(self, fields_list):
+        """At start partner_id is to be taken from active_id,
+        later defaults are
+        retained in the wizard record.
+        """
+        result = {}
+        result = super(PurchaseWizard, self).default_get(
+            fields_list=fields_list
+        )
+        product_model = self.env['product.product']
+        product_ids = self.env.context.get("active_ids", False)
+        if product_ids:
+            product = product_model.browse(product_ids[0])
+            supplier = product.product_tmpl_id.seller_ids[0]
+            result["product"] = product.id
+            if supplier:
+                partner = supplier.name
+                result["supplier"] = supplier.id
+                result["name"] = partner.name
+                result["stock_period_min"] = partner.stock_period_min
+                result["stock_period_max"] = partner.stock_period_max
+                result["stock_avl"] = product.virtual_available
+                today = date.today().strftime(DSDF)
+                result['ultimate_purchase'] = product.ultimate_purchase
+                pending_rfqs = self.env['purchase.order'].search(
+                    [("partner_id", '=', supplier.name.id),
+                     ("state", 'in', ['draft', 'sent'])]
+                ).ids
+                result['pending_rfq_lines'] = self.env[
+                    'purchase.order.line'].search(
+                        [('order_id', 'in', pending_rfqs),
+                         ('product_id', '=', product.id)]).ids
+                if partner.ultimate_purchase < today:
+                    result["ultimate_purchase_to"] = today
+        # TODO CHECK RFQ/PO MANAGMENT FROM PARTNER.
+        return result
+
+    def _get_qty(self, product, supplier, stock_period_max):
+        # ADD A DEFAULT STOCK PERIOD MAX OF 0.01 IN CASE IT IS 0.00
+        # the reason for this is that if a product has never been sold (new) or
+        # hasn't been sold recently (in period, seasonal products) we need a
+        # minimum to have a min order.
+        qty = float_round(
+            product.turnover_average * stock_period_max -
+            product.virtual_available, 0)
+        purchase_multiple = supplier.purchase_multiple
+        if purchase_multiple == 0:
+            purchase_multiple = 1
+        qty = int((qty + purchase_multiple - 1) / purchase_multiple)
```

Comment:
if you are expecting an integer division here, can you use the '//' operator? This will make the porting to python3 easier.